### PR TITLE
ci: add stale bot

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong Dai <loong.dai@intel.com>

### CODE REVIEW

**Review by arkodg — DISMISSED** [review]
looks like envoy has this CI as well, sgtm, thanks !

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      issues: write  # for actions/stale to close stale issues
+      pull-requests: write  # for actions/stale to close stale PRs
+    name: Prune Stale
+    runs-on: ubuntu-latest
+    # do not run it in forked repos
+    if: github.repository == 'envoyproxy/gateway'
+
+    steps:
+    - name: Prune Stale
+      uses: actions/stale@428b0ed64d74596742200dc98805bb90f39ef9f0
```
[the remark this answers is not in the document]
Comment (daixiang0) [reply .github/workflows/stale.yml:18]: Pin version with SHA, `v5` may change and introduce extra problem.

It is same with Envoy, feel free to change to `v5` if insist.

Comment (arkodg) [inline .github/workflows/stale.yml:18]:
```suggestion
      uses: actions/stale@v5
```

Comment (daixiang0) [reply .github/workflows/stale.yml]: Updated, please review again.

**Review by arkodg — APPROVED** [review]
🚀

**Pull request merged**